tests/utils/test_credentials.py

```diff
@@ -99,3 +102,29 @@ def test_read_mlflow_creds_env_takes_precedence_over_file(tmp_path, monkeypatch)
         creds = read_mlflow_creds()
         assert creds.username == "username_env"
         assert creds.password == "password_env"
+
+
+def test_mlflow_login(tmp_path):
+    with patch("builtins.input") as mock_input, patch("getpass.getpass") as mock_getpass:
+        mock_input.side_effect = ["https://community.cloud.databricks.com/", "weirdmouse"]
+        mock_getpass.side_effect = ["dummypassword"]
+
+        os.environ["DATABRICKS_CONFIG_FILE"] = f"{tmp_path}/.databrickscfg"
+
+        with patch("databricks.sdk.WorkspaceClient") as mock_workspace_client:
+
+            class FakeWorkspaceClient:
+                class FakeClusters:
+                    def list(self):
+                        return ["cluster1"]
+
+                def __init__(self):
+                    self.clusters = FakeWorkspaceClient.FakeClusters()
+
+            mock_workspace_client.side_effect = [
+                MlflowException("Error"),
+                FakeWorkspaceClient(),
+            ]
+
+            login("databricks")
+    del os.environ["DATABRICKS_CONFIG_FILE"]
```

Comment:
This line isn't called if an error occurs before this line. `monkeypatch.setenv` resets the environment variable even when an error occurs.